ec2api/api/route_table.py

```diff
@@ -480,18 +480,26 @@ def _update_routes_in_associated_subnets(context, route_table, cleaner,
 
 def _update_subnet_host_routes(context, subnet, route_table, cleaner=None,
                                rollback_route_table_object=None,
-                               router_objects=None, neutron=None):
+                               router_objects=None, neutron=None,
+                               vpc_route_revert=False, add_vpc_route=False):
     neutron = neutron or clients.neutron(context)
     os_subnet = neutron.show_subnet(subnet['os_id'])['subnet']
     gateway_ip = str(netaddr.IPAddress(
         netaddr.IPNetwork(os_subnet['cidr']).first + 1))
-    host_routes = _get_subnet_host_routes(context, route_table, gateway_ip,
-                                          router_objects)
+    if add_vpc_route == False:
+        host_routes = _get_subnet_host_routes(context, route_table, gateway_ip,
+                                              router_objects)
+    else:
+        host_routes = _get_subnet_host_routes(context, route_table, gateway_ip,
+                                              router_objects, True)
     neutron.update_subnet(subnet['os_id'],
                           {'subnet': {'host_routes': host_routes}})
     if cleaner and rollback_route_table_object:
         cleaner.addCleanup(_update_subnet_host_routes, context, subnet,
                            rollback_route_table_object)
+    elif vpc_route_revert == True:
+        cleaner.addCleanup(_update_subnet_host_routes, context, subnet,
+                           route_table, None, None, None, None, False, True)
 
```

Comment:
This will add a method in cleaner. However cleaner will be called only when there is an exception. In my opinion this cleanup method will not be called as there is no exception raised. Please let me know if you think otherwise.